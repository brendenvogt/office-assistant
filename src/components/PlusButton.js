import * as React from 'react';
import {
  View,
  Image,
  TouchableOpacity,
} from 'react-native-ui-lib';

import { colors } from '../styles';

export default function PlusButton(props): React.Node {
  return (
    <TouchableOpacity
      style={{
        position: 'absolute',
        right: 20,
        bottom: 20,
      }}
      testID="add-new-notification"
      {...props}
    >
      <View
        centerV
        centerH
        style={{
          backgroundColor: colors.darkBlue,
          height: 50,
          width: 50,
          borderRadius: 25,
        }}
      >
        <Image
          style={{
            width: 14,
            height: 14,
          }}
          assetName="plus"
          assetGroup="icons"
          resizeMode="contain"
        />
      </View>
    </TouchableOpacity>
  );
}
