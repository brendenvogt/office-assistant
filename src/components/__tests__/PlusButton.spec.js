/* eslint-disable no-undef */
import React from 'react';
import Enzyme from 'enzyme';

import PlusButton from '../PlusButton';

describe('Testing ToggleButton component', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <PlusButton
        onPress={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
