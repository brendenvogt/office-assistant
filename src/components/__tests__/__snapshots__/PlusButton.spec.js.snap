// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ToggleButton component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlusButton
    onPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        centerH={true}
        centerV={true}
        style={
          Object {
            "backgroundColor": "#2B313F",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          }
        }
        useNativeDriver={true}
      >
        <Image
          assetGroup="icons"
          assetName="plus"
          resizeMode="contain"
          style={
            Object {
              "height": 14,
              "with": 14,
            }
          }
        />
      </View>,
      "onPress": [Function],
      "style": Object {
        "bottom": 20,
        "position": "absolute",
        "right": 20,
      },
      "testID": "add-new-notification",
      "useNativeDriver": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centerH": true,
        "centerV": true,
        "children": <Image
          assetGroup="icons"
          assetName="plus"
          resizeMode="contain"
          style={
            Object {
              "height": 14,
              "with": 14,
            }
          }
        />,
        "style": Object {
          "backgroundColor": "#2B313F",
          "borderRadius": 25,
          "height": 50,
          "width": 50,
        },
        "useNativeDriver": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "assetGroup": "icons",
          "assetName": "plus",
          "resizeMode": "contain",
          "style": Object {
            "height": 14,
            "with": 14,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          centerH={true}
          centerV={true}
          style={
            Object {
              "backgroundColor": "#2B313F",
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            }
          }
          useNativeDriver={true}
        >
          <Image
            assetGroup="icons"
            assetName="plus"
            resizeMode="contain"
            style={
              Object {
                "height": 14,
                "with": 14,
              }
            }
          />
        </View>,
        "onPress": [Function],
        "style": Object {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
        },
        "testID": "add-new-notification",
        "useNativeDriver": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "centerH": true,
          "centerV": true,
          "children": <Image
            assetGroup="icons"
            assetName="plus"
            resizeMode="contain"
            style={
              Object {
                "height": 14,
                "with": 14,
              }
            }
          />,
          "style": Object {
            "backgroundColor": "#2B313F",
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          "useNativeDriver": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "assetGroup": "icons",
            "assetName": "plus",
            "resizeMode": "contain",
            "style": Object {
              "height": 14,
              "with": 14,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
